src/main.zig

```diff
@@ -1250,6 +1256,17 @@ fn buildOutputType(
                     .framework => try frameworks.append(it.only_arg),
                     .nostdlibinc => want_native_include_dirs = false,
                     .strip => strip = true,
+                    .treat_as_language => {
+                        if (mem.eql(u8, it.only_arg, "c") or mem.eql(u8, it.only_arg, "c-header")) {
+                            treat_as_file_ext = .c;
+                        } else if (mem.eql(u8, it.only_arg, "c++")) {
+                            treat_as_file_ext = .cpp;
+                        } else {
+                            treat_as_file_ext = .unknown;
+                            fatal("unsupported value '{s}' for -x or --language option", .{it.only_arg});
```

Comment:
Not true, there are many [other](https://github.com/llvm/llvm-project/blob/main/clang/include/clang/Driver/Types.def) valid cases that are not handled.

Reply:
> are you suggesting I change the error message or support all the valid options?

The latter, adding more valid extensions is also good for plain `zig` as, at the moment, it will outright reject any objective-c file.